cub/test/catch2_test_device_reduce_env_api.cu

```diff
@@ -15,108 +15,284 @@
 
 C2H_TEST("cub::DeviceReduce::Reduce accepts determinism requirements", "[reduce][env]")
 {
-  // TODO(gevtushenko): replace `run_to_run` with `gpu_to_gpu` once RFA unwraps contiguous iterators
-
   // example-begin reduce-env-determinism
   auto op     = cuda::std::plus{};
-  auto input  = c2h::device_vector<float>{0.0f, 1.0f, 2.0f, 3.0f};
-  auto output = c2h::device_vector<float>(1);
+  auto input  = thrust::device_vector<float>{0.0f, 1.0f, 2.0f, 3.0f};
+  auto output = thrust::device_vector<float>(1);
   auto init   = 0.0f;
 
+  // todo(srinivas): replace with gpu_to_gpu once offset size restriction is relaxed
   auto env = cuda::execution::require(cuda::execution::determinism::run_to_run);
 
-  cub::DeviceReduce::Reduce(input.begin(), output.begin(), input.size(), op, init, env);
+  auto error = cub::DeviceReduce::Reduce(input.begin(), output.begin(), input.size(), op, init, env);
 
-  c2h::device_vector<float> expected{6.0f};
+  thrust::device_vector<float> expected{6.0f};
   // example-end reduce-env-determinism
 
+  REQUIRE(error == cudaSuccess);
   REQUIRE(output == expected);
 }
 
 C2H_TEST("cub::DeviceReduce::Reduce accepts not_guaranteed determinism requirements", "[reduce][env]")
 {
   // example-begin reduce-env-non-determinism
   auto op     = cuda::std::plus{};
-  auto input  = c2h::device_vector<float>{0.0f, 1.0f, 2.0f, 3.0f};
-  auto output = c2h::device_vector<float>(1);
+  auto input  = thrust::device_vector<float>{0.0f, 1.0f, 2.0f, 3.0f};
+  auto output = thrust::device_vector<float>(1);
   auto init   = 0.0f;
 
   auto env = cuda::execution::require(cuda::execution::determinism::not_guaranteed);
 
-  cub::DeviceReduce::Reduce(input.begin(), output.begin(), input.size(), op, init, env);
+  auto error = cub::DeviceReduce::Reduce(input.begin(), output.begin(), input.size(), op, init, env);
 
-  c2h::device_vector<float> expected{6.0f};
+  thrust::device_vector<float> expected{6.0f};
   // example-end reduce-env-non-determinism
 
+  REQUIRE(error == cudaSuccess);
```

Comment:
Suggestion: should we also add to the example code between begin and end on what the proper way to check the error code is? I think this is important now that we added `nodiscard`. This also applies to the other tests in this file